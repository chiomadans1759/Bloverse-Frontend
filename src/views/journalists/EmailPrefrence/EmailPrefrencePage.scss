#manage-email {
  .small {
    font-size: rem(12px);
    color: #868686;
  }

  .title {
    font-size: rem(16px);
    color: #565656;
  }

  .btn-toggle {
    padding: 0;
    position: relative;
    border: rem(1px) solid rgba(0,0,0,0.15);
    height: 1.625rem;
    width: 3rem;
    background: #b7b7b7;
    border-radius: 0.125rem !important;
    &.btn-sm:focus {
      outline: none;
    }
  }

  .footer-container {
    position: fixed;
    bottom: 0;
    color: #B7B7B7;
    line-height: rem(40px);
  }
}

.btn-toggle.btn-sm {
  &.focus, &:focus.active, &.focus.active {
    outline: none;
  }
}

#manage-email .btn-toggle.btn-sm:before, .btn-toggle.btn-sm:after {
  line-height: 1.5rem;
  width: 0.5rem;
  text-align: center;
  font-weight: 600;
  font-size: 0.55rem;
  text-transform: uppercase;
  letter-spacing: 0.1rem;
  position: absolute;
  bottom: 0;
  transition: opacity 0.25s;
}

#manage-email .btn-toggle.btn-sm {
  &:before {
    content: 'Off';
    left: 1rem;
    text-transform: capitalize;
  }
  &:after {
    content: 'On';
    right: -0.5rem;
    opacity: 0.5;
    text-transform: capitalize;
  }
  > .square {
    position: absolute;
    top: 0.1875rem;
    left: 0.14rem;
    width: 1.125rem;
    height: 1.125rem;
    background: #ffffff;
    transition: left 0.25s;
  }
  &.active {
    transition: 0.25s;
    background: #096DD9;
    > .square {
      left: 1.6875rem;
      transition: left 0.25s;
    }
    &:before {
      opacity: 0.5;
    }
    &:after {
      opacity: 1;
    }
  }
  &.btn-sm:before {
    line-height: -0.5rem;
    color: #ffffff;
    letter-spacing: 0.1rem;
    left: 0.4125rem;
    width: 2.325rem;
  }
}

.btn-toggle.btn-sm.btn-sm:after {
  line-height: -0.5rem;
  color: #ffffff;
  letter-spacing: 0.1rem;
  left: 0.4125rem;
  width: 2.325rem;
}

#manage-email {
  .btn-toggle.btn-sm.btn-sm {
    &:before {
      text-align: right;
    }
    &:after {
      text-align: left;
      opacity: 0;
    }
    &.active {
      &:before {
        opacity: 0;
      }
      &:after {
        opacity: 1;
      }
    }
  }
  .hr {
    margin-top: 1rem;
    margin-bottom: 1rem;
    border: 0;
    border-top: 0.09rem solid rgba(0, 0, 0, 0.1);
    margin-left: -1.5rem;
    margin-right: -1.5rem;
  }
}
