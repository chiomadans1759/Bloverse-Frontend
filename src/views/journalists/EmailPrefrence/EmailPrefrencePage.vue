<template>
  <main id="manage-email">
    <navbar
      :showBackArrow="showBackArrow"
      :showNavigations="showNavigations"
      customClass="back-nav"
    />
    <div class="container py-4">
      <div class="offset-md-3 col-md-6">
        <div class="row">
          <h1>
            <strong>Manage Email Preferences</strong>
          </h1>
        </div>
        <div class="row pt-5">
          <div class="card p-4">
            <div class="pt-3 d-flex justify-content-between align-items-center pb-3">
              <div class="pr-2">
                <p class="text-capitalize title">Bloverse Article Digest</p>
                <p class="text small">Discover top, new articles based on your selected categories.</p>
              </div>
              <div>
                <button
                  type="button"
                  class="btn btn-sm btn-toggle active"
                  data-toggle="button"
                  aria-pressed="true"
                  autocomplete="off"
                >
                  <div class="square"></div>
                </button>
              </div>
            </div>
            <div class="pt-3 d-flex justify-content-between align-items-center pb-3">
              <div class="pr-2">
                <p class="text-capitalize title">Recommendations</p>
                <p class="text small">Discover trending, related articles that might interest you.</p>
              </div>
              <div>
                <button
                  type="button"
                  class="btn btn-sm btn-toggle active"
                  data-toggle="button"
                  aria-pressed="true"
                  autocomplete="on"
                >
                  <div class="square"></div>
                </button>
              </div>
            </div>
            <div class="pt-3 d-flex justify-content-between align-items-center pb-3">
              <div class="pr-2">
                <p class="text-capitalize title">Photo Contest</p>
                <p class="text small">Discover top, new articles based on your selected categories.</p>
              </div>
              <div>
                <button
                  type="button"
                  class="btn btn-sm btn-toggle active"
                  data-toggle="button"
                  aria-pressed="true"
                  autocomplete="on"
                >
                  <div class="square"></div>
                </button>
              </div>
            </div>
            <div class="pt-3 d-flex justify-content-between align-items-center pb-3">
              <div class="pr-2">
                <p class="text-capitalize title">News & Announcements</p>
                <p
                  class="text small"
                >Stay updated on new features and products thatâ€™ll improve your experience.</p>
              </div>
              <div>
                <button
                  type="button"
                  class="btn btn-sm btn-toggle active"
                  data-toggle="button"
                  aria-pressed="true"
                  autocomplete="off"
                >
                  <div class="square"></div>
                </button>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
    <div class="footer-container">
      <p>
        <span class="bloverse">bloverse</span> 2019. All Rights Reserved
      </p>
    </div>
  </main>
</template>

<script>
import Navbar from "@/components/Navbar/Navbar";

export default {
  name: "EmailSettings",
  components: { Navbar },
  data() {
    return {
      showNavigations: false,
      showBackArrow: true
    };
  }
};
</script>

<style lang="scss" scoped>
@import "./EmailPrefrencePage";
</style>
