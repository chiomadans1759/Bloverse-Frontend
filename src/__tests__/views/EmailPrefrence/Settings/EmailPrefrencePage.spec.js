import { mount, RouterLinkStub } from '@vue/test-utils';
import EmailPrefrence from '@/views/journalists/EmailPrefrence/EmailPrefrencePage.vue';

describe('Creator Email Settings Page', () => {
  it('should render properly', () => {
    const wrapper = mount(EmailPrefrence, {
      stubs: {
        RouterLink: RouterLinkStub
      }
    });
    expect(wrapper.element).toMatchSnapshot();
  });
});
