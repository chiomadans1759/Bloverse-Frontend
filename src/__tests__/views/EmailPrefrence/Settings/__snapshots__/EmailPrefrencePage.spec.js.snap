// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Creator Email Settings Page should render properly 1`] = `
<main
  id="manage-email"
>
  <header
    id="feeds-header"
  >
    <nav
      class="navbar navbar-expand-lg navbar-light px-5"
    >
      <div
        class="container-fluid back-nav"
      >
        <a
          class="backlink mr-4"
          href="/settings"
        >
          <i
            class="fal fa-long-arrow-left fa-2x back-icon"
          />
           
          <span
            class="back-text"
          >
            Back
          </span>
        </a>
         
        <a
          class="router-link"
        >
          <img
            class="nav-logo"
            src="@/assets/Bloverse.svg"
          />
           
          <!---->
        </a>
         
        <!---->
         
        <!---->
      </div>
    </nav>
     
    <main
      id="bl-modal"
    >
      <div
        aria-hidden="true"
        aria-labelledby="customModal"
        class="modal fade"
        id="feedback-modal"
        role="dialog"
        tabindex="-1"
      >
        <div
          class="modal-dialog modal-dialog-centered"
          role="document"
        >
          <div
            class="modal-content"
          >
            <div
              class="modal-header darkBackground"
            >
              <!---->
               
              <div>
                <h6>
                  <strong
                    class="alt-title"
                  >
                    Send Us Some Feedback
                  </strong>
                </h6>
                 
                <h6
                  class="mt-2 sub-title"
                >
                  Found a bug? Have a suggestion?. Fill out the form below and we'll take a look!
                </h6>
              </div>
               
              <button
                aria-label="Close"
                class="close"
                data-dismiss="modal"
                type="button"
              >
                <span
                  aria-hidden="true"
                >
                  ×
                </span>
              </button>
            </div>
             
            <div
              class="modal-body"
            >
              <div
                class="bl-modal-content"
              >
                <p>
                  Enter your feedback here!
                </p>
                 
                <textarea
                  class="form-control mb-2 p-1"
                />
                 
                <label
                  class="ml-1 mr-3"
                >
                  <input
                    name="feedback-type"
                    type="radio"
                  />
                    Bug
                </label>
                 
                <label
                  class="mr-3"
                >
                  <input
                    name="feedback-type"
                    type="radio"
                  />
                    Comment
                </label>
                 
                <label
                  class="mr-3"
                >
                  <input
                    name="feedback-type"
                    type="radio"
                  />
                    Other
                </label>
              </div>
            </div>
             
            <div
              class="modal-footer"
            >
              <button
                class="btn btn-outline-primary"
                data-dismiss="modal"
                type="button"
              >
                Close
              </button>
               
              <button
                class="btn btn-primary"
                type="button"
              >
                Send Feedback
              </button>
            </div>
          </div>
        </div>
      </div>
    </main>
  </header>
   
  <div
    class="container py-4"
  >
    <div
      class="offset-md-3 col-md-6"
    >
      <div
        class="row"
      >
        <h1>
          <strong>
            Manage Email Preferences
          </strong>
        </h1>
      </div>
       
      <div
        class="row pt-5"
      >
        <div
          class="card p-4"
        >
          <div
            class="pt-3 d-flex justify-content-between align-items-center pb-3"
          >
            <div
              class="pr-2"
            >
              <p
                class="text-capitalize title"
              >
                Bloverse Article Digest
              </p>
               
              <p
                class="text small"
              >
                Discover top, new articles based on your selected categories.
              </p>
            </div>
             
            <div>
              <button
                aria-pressed="true"
                autocomplete="off"
                class="btn btn-sm btn-toggle active"
                data-toggle="button"
                type="button"
              >
                <div
                  class="square"
                />
              </button>
            </div>
          </div>
           
          <div
            class="pt-3 d-flex justify-content-between align-items-center pb-3"
          >
            <div
              class="pr-2"
            >
              <p
                class="text-capitalize title"
              >
                Recommendations
              </p>
               
              <p
                class="text small"
              >
                Discover trending, related articles that might interest you.
              </p>
            </div>
             
            <div>
              <button
                aria-pressed="true"
                autocomplete="on"
                class="btn btn-sm btn-toggle active"
                data-toggle="button"
                type="button"
              >
                <div
                  class="square"
                />
              </button>
            </div>
          </div>
           
          <div
            class="pt-3 d-flex justify-content-between align-items-center pb-3"
          >
            <div
              class="pr-2"
            >
              <p
                class="text-capitalize title"
              >
                Photo Contest
              </p>
               
              <p
                class="text small"
              >
                Discover top, new articles based on your selected categories.
              </p>
            </div>
             
            <div>
              <button
                aria-pressed="true"
                autocomplete="on"
                class="btn btn-sm btn-toggle active"
                data-toggle="button"
                type="button"
              >
                <div
                  class="square"
                />
              </button>
            </div>
          </div>
           
          <div
            class="pt-3 d-flex justify-content-between align-items-center pb-3"
          >
            <div
              class="pr-2"
            >
              <p
                class="text-capitalize title"
              >
                News & Announcements
              </p>
               
              <p
                class="text small"
              >
                Stay updated on new features and products that’ll improve your experience.
              </p>
            </div>
             
            <div>
              <button
                aria-pressed="true"
                autocomplete="off"
                class="btn btn-sm btn-toggle active"
                data-toggle="button"
                type="button"
              >
                <div
                  class="square"
                />
              </button>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
   
  <div
    class="footer-container"
  >
    <p>
      <span
        class="bloverse"
      >
        bloverse
      </span>
       2019. All Rights Reserved
    
    </p>
  </div>
</main>
`;
